src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,532 @@
+from jax.scipy.linalg import block_diag
+
+import jaxsim.api as js
+import jaxsim.typing as jtp
+from jaxsim import math
+from jaxsim.api.common import VelRepr
+from jaxsim.terrain.terrain import FlatTerrain, Terrain
+
+from .common import ContactModel, ContactsParams, ContactsState
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContactParams(ContactsParams):
+    """Parameters of the rigid contacts model."""
+
+    # Static friction coefficient
+    mu: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.5, dtype=float)
+    )
+
+    # Baumgarte proportional term
+    K: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.0, dtype=float)
+    )
+
+    # Baumgarte derivative term
+    D: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.0, dtype=float)
+    )
+
+    def __hash__(self) -> int:
+        from jaxsim.utils.wrappers import HashedNumpyArray
+
+        return hash(
+            (
+                HashedNumpyArray.hash_of_array(self.mu),
+                HashedNumpyArray.hash_of_array(self.K),
+                HashedNumpyArray.hash_of_array(self.D),
+            )
+        )
+
+    def __eq__(self, other: RigidContactParams) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build(
+        inactive_points_prev: jtp.Vector,
+        mu: jtp.Float = 0.5,
+        K: jtp.Float = 0.0,
+        D: jtp.Float = 0.0,
+    ) -> RigidContactParams:
+        """Create a `RigidContactParams` instance"""
+        return RigidContactParams(mu=mu, K=K, D=D)
+
+    @staticmethod
+    def build_from_jaxsim_model(
+        model: js.model.JaxSimModel,
+        *,
+        static_friction_coefficient: jtp.Float = 0.5,
+        K: jtp.Float = 0.0,
+        D: jtp.Float = 0.0,
+    ) -> RigidContactParams:
+        """Build a `RigidContactParams` instance from a `JaxSimModel`."""
+
+        return RigidContactParams.build(
+            mu=static_friction_coefficient,
+            K=K,
+            D=D,
+        )
+
+    def valid(self) -> bool:
+        return bool(
+            jnp.all(self.mu >= 0.0)
+            and jnp.all(self.K >= 0.0)
+            and jnp.all(self.D >= 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContactsState(ContactsState):
+    """Class storing the state of the rigid contacts model."""
+
+    def __eq__(self, other: RigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build_from_jaxsim_model(
+        model: js.model.JaxSimModel | None = None,
+    ) -> RigidContactsState:
+        """Build a `RigidContactsState` instance from a `JaxSimModel`."""
+        return RigidContactsState.build()
+
+    @staticmethod
+    def build() -> RigidContactsState:
+        """Create a `RigidContactsState` instance"""
+
+        return RigidContactsState()
+
+    @staticmethod
+    def zero(model: js.model.JaxSimModel) -> RigidContactsState:
```

Comment:
Unused argument

```suggestion
    def zero() -> RigidContactsState:
```

Reply:
Done in 010ec988da9f7eb5f5fb1863effbcfc6c8f9fb58